crates/uv-cli/src/lib.rs

```diff
@@ -134,20 +134,39 @@ pub struct TopLevelArgs {
 #[command(next_help_heading = "Global options", next_display_order = 1000)]
 #[allow(clippy::struct_excessive_bools)]
 pub struct GlobalArgs {
-    /// Whether to prefer uv-managed or system Python installations.
+    #[arg(
+        global = true,
+        long,
+        help_heading = "Python options",
+        display_order = 700,
+        env = EnvVars::UV_PYTHON_PREFERENCE,
+        hide = true
+    )]
+    pub python_preference: Option<PythonPreference>,
+
+    /// Whether to require uv-managed Python installations.
     ///
     /// By default, uv prefers using Python versions it manages. However, it
     /// will use system Python installations if a uv-managed Python is not
-    /// installed. This option allows prioritizing or ignoring system Python
-    /// installations.
+    /// installed. This option disables this system Python installation fallback.
     #[arg(
         global = true,
         long,
         help_heading = "Python options",
-        display_order = 700,
-        env = EnvVars::UV_PYTHON_PREFERENCE
+        conflicts_with = "no_managed_python",
+        conflicts_with = "python_preference"
     )]
-    pub python_preference: Option<PythonPreference>,
+    pub managed_python: bool,
+
+    /// Whether to disable uv-managed Python installations.
```

Comment:
I would re-phrase (I think we use "Whether" when we only show one of the options)

```suggestion
    /// Disable use of uv-managed Python distributions.
```
I think we could say a bit more here, too, like

> Instead, uv will search for a suitable Python installation on the system.